public/controllers/management/components/management/common/actions-buttons.tsx

```diff
@@ -0,0 +1,186 @@
+    return results;
+  } catch (error) {
+    throw error;
+  }
+}
+
+const getReadPermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+const getUpdatePermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:update`,
+      resource: permissionResource('*'),
+    },
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+// Add new rule button
+export const AddNewFileButton = ({ section, updateAddingFile }) => (
+  <>{
+    section !== 'lists' &&
+    <WzButtonPermissions
+      permissions={getUpdatePermissionsFiles(section)}
+      buttonType="empty"
+      iconType="plusInCircle"
+      onClick={() =>
+        updateAddingFile({
+          name: '',
+          content: '<!-- Modify it at your will. -->',
+          path: `etc/${section}`,
+        })
+      }
+    >
+      {`Add new ${section} file`}
+    </WzButtonPermissions>
+  }</>
+)
+
+//Add new CDB list button
+export const AddNewCdbListButton = (({ section, updateListContent }) => {
+  return <>
+      <WzButtonPermissions
+        buttonType="empty"
+        permissions={getUpdatePermissionsFiles(section)}
+        iconType="plusInCircle"
+        onClick={() =>
+          updateListContent({
+            name: false,
+            content: '',
+            path: 'etc/lists',
+          })
+        }
+      >
+        {`Add new ${section} file`}
+      </WzButtonPermissions>
+  </>
+});
+
+// Manage files
+export const ManageFiles = (({ section, showingFiles, ...props }) => {
+  
+  /**
+ * Toggle between files and rules or decoders
+ */
+  const toggleFiles = async () => {
+    try {
+      props.toggleShowFiles(!showingFiles);
+    } catch (error) {
+      const options = {
+        context: 'ActionButtons.toggleFiles',
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        store: false,
+        error: {
+          error: error,
+          message: `Error toggling to files: ${error.message || error}`,
+          title: error.name || error,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
+    }
+  }
+  return (
+    <>
+      {section !== 'lists' &&
+        <WzButtonPermissions
+          buttonType="empty"
+          permissions={getUpdatePermissionsFiles(section)}
+          iconType={showingFiles ? 'apmTrace' : 'folderClosed'}
+          onClick={async () => await toggleFiles()}
+        >
+          {showingFiles ? `Manage ${section}` : `Manage ${section} files`}
+        </WzButtonPermissions>}
+    </>
+  )
+});
+
+const uploadFile = async (files, resource, overwrite) => {
+  try {
+    const result = await uploadFiles(files, resource, overwrite);
+    return result;
+  } catch (error) {
+    return error;
```

Comment:
**thought**: Is it managed that the error can be returned? It looks strange to me.